### PR TITLE
feat(hierarchical): add prefix path option to hierarchical facet

### PR DESCRIPTION
**Pull request by maxiloc**
Here the implementation for #251

### CODE REVIEW

**Comment by maxiloc** [comment]
cc @vvo

File: src/SearchParameters/index.js [hunk 1]
```diff
@@ -1380,6 +1380,15 @@ SearchParameters.prototype = {
   },
 
   /**
+   * Helper function to get the hierarchicalFacet prefix path or the default one (``)
```
Comment (pixelastic) [inline src/SearchParameters/index.js]: What is inside the parenthesis?

File: src/SearchParameters/index.js [hunk 2]
```diff
@@ -1380,6 +1380,15 @@ SearchParameters.prototype = {
...
+   * @param  {object} hierarchicalFacet
+   * @return {string} returns the hierarchicalFacet.prefixPath or `` as default
```
Comment (pixelastic) [inline src/SearchParameters/index.js]: It seems to return `null`, not an empty string

**Comment by maxiloc** [comment]
@pixelastic fixed

File: src/SearchResults/generate-hierarchical-tree.js [hunk 3]
```diff
@@ -59,7 +61,10 @@ function generateHierarchicalTree(sortBy, hierarchicalSeparator, currentRefineme
       //
       // If parent refinement is `beers`, then we do not want to have `bières > Belges`
       // showing up
-      var onlyMatchingValuesFn = filterFacetValues(parent.path, currentRefinement, hierarchicalSeparator);
+
+      var onlyMatchingValuesFn = filterFacetValues(parent.path || hierarchicalPrefixPath,
```
Comment (vvo) [inline src/SearchResults/generate-hierarchical-tree.js]: This test fails is parent.path is an empty string, or '0'. Can you compute it with the right default values/undefined?

**Comment by vvo** [comment]
Test failing on IE8, expected. Otherwise good

**Pull request merged**